File: src/mod10_design_pattern/lesson8_bridge/java/DebitAccount.java
package mod10_design_pattern.lesson8_bridge.java;

public class DebitAccount extends Account {

    private double balance;

    public DebitAccount(Currency currency) {
        super(currency);
    }

    @Override
    public String getAccountType() {
        return "Debit account with " + currency.getCurrencyName();
    }

    public void withdraw(double amount) {
        if (balance >= amount) {
            balance -= amount;
            System.out.println("Withdraw: " + amount + " " + currency.getCurrencyName());
        } else {
            System.out.println("Not enough funds for withdrawal!");
        }
    }

    public void deposit(double amount) {
        balance += amount;
        System.out.println("Deposit: " + amount + " " + currency.getCurrencyName());
    }

    public void displayBalance(){
        System.out.println("Balance: "+ getBalanceConvertedToGalleons(balance)+ " Galleons.");
    }

}

File: src/mod10_design_pattern/lesson8_bridge/kotlin/DebitAccount.kt
package mod10_design_pattern.lesson8_bridge.kotlin

class DebitAccount(
    private val currency: Currency
) : Account(currency) {

    private var balance = 0.0

    override fun getAccountType() = "Debit account in ${currency.getCurrencyName()}"

    fun deposit(amount: Double) {
        balance += amount
        println("Deposit: $amount ${currency.getCurrencyName()}")
    }

    fun withdraw(amount: Double) {
        if (balance >= amount) {
            balance -= amount
            println("Withdraw: $amount ${currency.getCurrencyName()}")
        } else {
            println("Not enough fund for withdraw!")
        }
    }

    fun dislayBalance(){
        println("Balance: ${getBalanceConvertedToGalleons(balance)} Galleons.")
    }
}